lib/ansible/galaxy/role.py

```diff
@@ -379,7 +379,14 @@ def install(self):
                                     if n_part != '..' and not n_part.startswith('~') and '$' not in n_part:
                                         n_final_parts.append(n_part)
                                 member.name = os.path.join(*n_final_parts)
-                                role_tar_file.extract(member, to_native(self.path))
+                                maybe_filter_kwargs = {
+                                    'filter': 'data',
+                                } if hasattr(tarfile, 'data_filter') else {}
+                                # deprecated: description='extract fallback without filter' python_version='3.11'
+                                role_tar_file.extract(
+                                    member, to_native(self.path),
+                                    **maybe_filter_kwargs,
+                                )
```

Comment:
Can you switch this back to the previous if/else approach? This form is more verbose. At least for me, it's also more difficult to read.